pptx_blueprint/__init__.py

```diff
@@ -54,7 +54,27 @@ def _find_shapes(self, label: str) -> Iterable[BaseShape]:
         Args:
             label (str): label of the placeholder (without curly braces)
         """
-        pass
+        slide_number, tag_name = label.split(":")
+        matched_shapes = []
+
+        def _find_shapes_in_slide(slide):
+            for shape in slide.shapes:
+                if shape.text == f'{{{tag_name}}}':
+                    matched_shapes.append(shape)
+
+        if slide_number == '*':
+            for slide in self._presentation.slides:
```

Comment:
after refactoring to give the slide numbers as an iterable, you could use 'map' for this iteration